### ISSUE TITLE
Initial weight not configured correctly

### ISSUE THREAD

**Issue by mhyllander** [jcmoraisjr/haproxy-ingress#741]
I have set `initial-weight` to "100" on an Ingress, but the `server-template` in the generated config has `weight 1`. 

When I configure:

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    ingress.kubernetes.io/agent-check-port: "9991"
    ingress.kubernetes.io/initial-weight: "100"
    ingress.kubernetes.io/timeout-tunnel: 60s
    ingress.kubernetes.io/use-resolver: kubernetes
```

I get the following config in haproxy.cfg:

```
backend platform_re-worker-service_websocket
    mode http
    balance roundrobin
    timeout tunnel 60s
    acl https-request ssl_fc
    http-request redirect scheme https if !https-request
    option forwardfor
    http-response set-header Strict-Transport-Security "max-age=15768000"
    server-template srv 100 _websocket._tcp.re-worker-service.platform.svc.cluster.local resolvers kubernetes resolve-prefer ipv4 init-addr none weight 1 check inter 2s agent-check agent-port 9991
```

Notice that it has "weight 1".

I expected it to be "weight 100".

**Comment by jcmoraisjr** [comment]
Hi, I can confirm this behavior. The initial weight config is being written before it can be read from the ingress resource. You can work around it annotating the service resource instead.

**Comment by mhyllander** [comment]
Thanks. I guess I'll have to split the service in two, since I don't want the configuration to apply to all server endpoints.

**Comment by jcmoraisjr** [comment]
Yep, already fixed and planning a new round of releases this weekend. Maybe worth to wait a little more. Btw thanks for reporting.

**Comment by jcmoraisjr** [comment]
Fixed in v0.10.5, v0.11.4 and v0.12.1. Closing.
